cylc/uiserver/handlers.py

```diff
@@ -135,9 +183,13 @@ async def execute(self, *args, **kwargs):
             **kwargs,
         )
 
+    @web.authenticated
+    @async_authorised
```

Comment:
I think we copied this code from a Tornado pull request. And back then, we didn't add the `web.authenticated` as `prepare` had these already. Now I suspect `web.authenticated` and `async_authorised` are executed 3 times?

`execute` is called by `graphene-tornado`'s `TornadoGraphQLHandler` in `execute_graphql_request`.

Probably we can leave `web.authenticated` and `async_authorised` in `prepare` only?

Reply:
We can look into these multiple calls to authenticate/authorize later :+1:

Reply:
> Now I suspect web.authenticated and async_authorised are executed 3 times?

Yep, I expect you are right, I've been over cautious here to be safe.

I suggest we open an issue to come back to this for a later beta release as I hope there is a better way of going about it. There must be a way of making auth happen by default?

#### context

I couldn't easily tell what path requests would take through the code. When you are subclassing a handler, without digging through the code you have no idea what methods that handler implements, (get, post, delete, etc). My fear is that a subclassed handler we use will later implement a method we aren't expecting and open up an unauthorised route into the code.

So I just slapped these everywhere as auth is computationally cheep to perform but an unauthorised endpoint could be very very expensive in other ways.

I would really like a way to change the default so we authorise/authenticate all methods of all endpoints unless stated otherwise. Some other web frameworks use `@expose`-like decorators for similar reasons.

Reply:
Let's move it to a separate issue then. Thanks!

Reply:
Ok bumped - https://github.com/cylc/cylc-uiserver/issues/207